src/app/vm/vm-creation/vm-creation.component.ts

```diff
@@ -211,6 +228,23 @@ export class VmCreationComponent implements OnInit {
   }
 
   public deploy(): void {
+    this.templateTagService.getAgreement(this.formState.state.template)
```

Comment:
This is not correct way to handle confirmation Dialog
Please change it to call deployRequest once in subscribe